lib/macho/utils.rb

```diff
@@ -33,4 +33,16 @@ def self.magic32?(num)
   def self.magic64?(num)
     num == MH_MAGIC_64 || num == MH_CIGAM_64
   end
+
+  # @param num [Fixnum] the number being checked
+  # @return [Boolean] true if `num` is a valid little-endian magic number, false otherwise
+  def self.little_magic?(num)
+    num == FAT_CIGAM || num == MH_CIGAM || num == MH_CIGAM_64
```

Comment:
You can drop `FAT_CIGAM` here as the `fat_header` structure is *always* big endian, so for that particular structure you can hardcode the endianess as big. (The same applies to the `fat_arch` structure.)

What this also means is that `FAT_CIGAM` should be never used in this Ruby code. Since we read the magic number using the `N` format (32-bit unsigned big-endian integer), the magic number for a fat binary will *always* be `FAT_MAGIC`. This also means that `MachO.magic?` and `MachO.fat_magic?` need to be tweaked a bit. (The `FAT_CIGAM` constant mainly exists for compiled programs, so that they can easily detect whether the binary they are reading has the same endianess or whether all fields need to be byte-swapped during reading.)